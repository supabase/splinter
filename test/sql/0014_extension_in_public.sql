begin;
  set local search_path = '';

  -- 0 issues
  select * from lint."0014_extension_in_public";

  create extension ltree schema public;

  -- 1 issues
  select * from lint."0014_extension_in_public";

  -- resolve the problem
  alter extension ltree set schema extensions;
  
  -- 0 issues
  select * from lint."0014_extension_in_public";

rollback;
