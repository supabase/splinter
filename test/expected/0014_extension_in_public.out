begin;
  set local search_path = '';
  -- 0 issues
  select * from lint."0014_extension_in_public";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  create extension ltree schema public;
  -- 1 issues
  select * from lint."0014_extension_in_public";
        name         | level |  facing  | categories |                      description                       |                                      detail                                       |                                       remediation                                       |                          metadata                          |         cache_key         
---------------------+-------+----------+------------+--------------------------------------------------------+-----------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------+------------------------------------------------------------+---------------------------
 extension_in_public | WARN  | EXTERNAL | {SECURITY} | Detects extensions installed in the \`public\` schema. | Extension \`ltree\` is installed in the public schema. Move it to another schema. | https://supabase.com/docs/guides/database/database-linter?lint=0014_extension_in_public | {"name": "ltree", "type": "extension", "schema": "public"} | extension_in_public_ltree
(1 row)

  -- resolve the problem
  alter extension ltree set schema extensions;
  
  -- 0 issues
  select * from lint."0014_extension_in_public";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
